assertj-core/src/main/java/org/assertj/core/internal/Classes.java

```diff
@@ -170,6 +172,44 @@ public void assertIsAnnotation(AssertionInfo info, Class<?> actual) {
     if (!actual.isAnnotation()) throw failures.failure(info, shouldBeAnnotation(actual));
   }
 
+  /**
+   * Verifies that the actual {@code Class} is not a record.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the "actual" {@code Class}.
+   * @throws AssertionError if {@code actual} is {@code null}.
+   * @throws AssertionError if the actual {@code Class} is a record.
+   */
+  public void assertIsNotRecord(AssertionInfo info, Class<?> actual) {
+    assertNotNull(info, actual);
+    if (isRecord(actual)) throw failures.failure(info, shouldNotBeRecord(actual));
+  }
+
+  /**
+   * Verifies that the actual {@code Class} is a record.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the "actual" {@code Class}.
+   * @throws AssertionError if {@code actual} is {@code null}.
+   * @throws AssertionError if the actual {@code Class} is not a record.
+   */
+  public void assertIsRecord(AssertionInfo info, Class<?> actual) {
+    assertNotNull(info, actual);
+    if (!isRecord(actual)) throw failures.failure(info, shouldBeRecord(actual));
+  }
+
+  private boolean isRecord(Class<?> actual) {
+    try {
+      Method isRecordMethod = Class.class.getMethod("isRecord");
+      return (boolean) isRecordMethod.invoke(actual);
+    } catch (NoSuchMethodException e) {
+      // Definitely not a record if we're running on a JVM that doesn't support records
+      return false;
```

Comment:
If we go with reflection, we could deliver this as part of 3.x but we should also add an integration test using Java 17 as language level.

Would you mind creating a new `assertj-core-java-17` Maven module under https://github.com/assertj/assertj/tree/main/assertj-tests/assertj-integration-tests? You can tune the Java version in the corresponding POM via the `java.version` property, similar to what we have here:

https://github.com/assertj/assertj/blob/066cc49bb000d902745b726e1bf566cbdf703c30/assertj-tests/assertj-integration-tests/assertj-core-module-path/pom.xml#L15-L17

Reply:
Thanks @scordio, I've added a module as requested. I wasn't quite sure what naming conventions to follow for the test class that I was adding there, I called it `RecordTest` in the end to keep it simple like the other IT modules. I thought that was a reasonable approach since I don't imagine many more tests will be added to that module, but happy to revisit.